cypress/e2e/spec.cy.js

```diff
@@ -0,0 +1,22 @@
+describe('template spec', () => {
+  beforeEach(() => {
+    cy.visit('/')
+  })
+
+  it('Exibe a saudação "Hi, there" quando nenhum nome é fornecido', () => {
+    cy.get('input[data-testid="name"]').clear()
+    cy.contains('h2','Hi there!').should('be.visible')
+  })
+
+  it('Exibe a saudação "Hi, Joe" quando o nome é fornecido', () => {
+    cy.get('input[data-testid="name"]').clear().type('Joe')
+    cy.contains('h2','Hi Joe!').should('be.visible')
+  })
+
+  it('Abre e fecha o messenger ', () => {
+    cy.get('button[class*=Messenger]').first().should('be.visible').click()
```

Comment:
Recomendo usar um seletor mais legível aqui.

Algo assim:

```suggestion
    cy.get('button[aria-label="Open messenger"]').click()

```

A propósito, não há necessidade de verificar a visibilidade antes do clique visto que o Cypress já faz isso automaticamente.

Tal prática é útil em casos específicos, mas esses casos são excessões e não devem ser a regra.

Reply:
Que tal deixar espaços em branco entre as diferentes partes do testes (_Arrange_, _Act_, _Assert_)?

Ao separar as diferentes partes do teste a legibilidade do mesmo melhora e fica mais fácil identificar suas pré-condições, ações e resultados esperados.